### DISCUSSION TITLE
no such module

### DISCUSSION THREAD

**Discussion by IncredibleHohol** [wojciech-kulik/xcodebuild.nvim#166] [Q&A] [answered]
UPD: I found solution when re read complete guide, multiple projects not supported yet, so I modify my project.yml (delete target TestSDK) and now it's work

Hello! First of all, I'm completely new to iOS development and maybe my question will be silly, but I hope you can help me and this thread would help someone in future

I tried my nvim setup on another project and it's work like a charm, so we can exclude nvim setup from problem (at least I hope so)

I generate xcodeproj with xcodegen

my project.yml listed below

The problem: when I try to import my library into Application module classes, it says "no such module 'TestSDK'" and also LSP/autocompletion doesn't work
When I open project in Xcode same code works fine

I found out that problem is deeper, LSP in TestSDK module also doesn't work correctly 

In both scenarios (Xcode/nvim) I can build and run project without problem

```
name: TestSdkProject
options:
  bundleIdPrefix: com.test
  deploymentTarget:
    iOS: '15.0'
  
configurations:
  Debug:
    buildSettings:
      SWIFT_OPTIMIZATION_LEVEL: '-Onone'
      GENERATE_INFOPLIST_FILE: YES
  Release:
    buildSettings:
      SWIFT_OPTIMIZATION_LEVEL: '-Owholemodule'
      GENERATE_INFOPLIST_FILE: YES

schemes:
  TestSdkTestApp:
    build:
      targets:
        TestSdkTestApp: all
    run:
      buildConfiguration: Debug

targets:
  TestSdkTestApp:
    type: application
    platform: iOS
    sources: [Sources/TestSdkTestApp]
    dependencies:
      - target: TestSdk
    settings:
      base:
        INFOPLIST_FILE: Sources/TestSdkTestApp/Info.plist
        PRODUCT_BUNDLE_IDENTIFIER: com.test.TestSdkTestApp

  TestSdk:
    type: framework
    platform: iOS
    sources: [Sources/TestSdk]
    settings:
      base:
        GENERATE_INFOPLIST_FILE: YES
        BUILD_LIBRARY_FOR_DISTRIBUTION: YES
```

**Comment by wojciech-kulik** [comment]
Hmm, this plugin doesn't support multiple `xcodeproj` files. However, it works well with multiple targets.

Maybe try a clean build in Xcode and then run `xcode-build-server config -project TestSdkProject.xcodeproj -scheme TestSdkTestApp`

Btw. you use `configurations:` but based on the documentation it should be `configs:`. I'm not sure if it's gonna change anything.

**Comment by IncredibleHohol** [comment]
I decided to try chatgpt to help me with the configuration, I threw all the `xcodegen` documentation in there and got a new `project.yml`. Now it works fine, I can import the sdk module into the app module and the lsp browses the files correctly

Sorry to bother you. It became a classic case of "if you need to figure something out, just ask for help and you will quickly find a solution yourself"

Thanks again for the great plugin! No more suffering with wierd xcode behaviour

final project.yml
```
name: TestSdk

options:
  deploymentTarget:
    iOS: "15.0"
  settingPresets: all

configs:
  Debug: debug
  Release: release

targets:
  TestSdk:
    type: framework
    platform: iOS
    deploymentTarget: "15.0"
    sources:
      - path: Sources/TestSdk
    info:
      path: Sources/TestSdk/Info.plist
    settings:
      base:
        PRODUCT_BUNDLE_IDENTIFIER: com.test.testsdk
    dependencies: []

  TestSdkTestApp:
    type: application
    platform: iOS
    deploymentTarget: "15.0"
    sources:
      - path: Sources/TestSdkTestApp
    info:
      path: Sources/TestSdkTestApp/Info.plist
    settings:
      base:
        PRODUCT_BUNDLE_IDENTIFIER: com.test.testsdktestapp
    dependencies:
      - target: TestSdk

schemes:
  TestSdk:
    build:
      targets:
        TestSdk: all
    test:
      targets:
        - TestSdk

  TestSdkTestApp:
    build:
      targets:
        TestSdkTestApp: all
    run:
      config: Debug
```

**Reply by wojciech-kulik** [reply, marked as answer]
Great that you resolved your issue 🍻
